### PR TITLE
Cc3220sf host driver

### PR DESCRIPTION
**Pull request by GAnthony**
This PR imports the SimpleLink host driver v 1.40.00.03 (which communicates to the Network Coprocessor over dedicated SPI) plus its port to Zephyr for the CC3220SF_LaunchXL.

This is the first step to enable BSD socket offload over WiFi for the CC3220SF.

This is a replacement for the stale, now abandoned, PR #412, and is based on the (merged) SimpleLink SDK v 1.40.00.03 update PR #772.

### CODE REVIEW

**Comment by galak** [comment]
What's the issue with conflict on __errno?

**Comment by GAnthony** [comment]
SimpleLink defines its own __errno, but has a feature to use the porting OS's definition.   This patch actually should be using Zephyr's __errno definition.  CONFIG_ERRNO=y, so there should be no conflict.

**Comment by galak** [comment]
Can we than move the errno fixes out of the 'ext: simplelink: Add SimpleLink DPL porting layer to Zephyr' patch and into 'ext: simplelink: Enable build of the SimpleLink host driver.'?

**Comment by GAnthony** [comment]
Actually, that would be a bit difficult, as in order to enable use of Zephyr's __errno, the dpl_set_errno() function must be defined, which is part of the SimpleLink DPL porting layer, which comes in a later commit.

When staging the commits, it seemed logical to first simply enable the build of the host driver (with CONFIG_ERRNO=n, which defaults to using SimpleLink's __errno); then later, in the DPL porting layer commit, to actually hook into Zephyr's __errno (as it's part of the OS porting task), then set CONFIG_ERRNO=y.

**Comment by GAnthony** [comment]
So, to clarify, CONFIG_ERRNO=n is OK for the  'ext: simplelink: Enable build of the SimpleLink host driver.' commit, and creates no conflict.  It just defaults to using SimpleLink's __errno.  
The later commit, when DPL layer is ported to Zephyr, then enables using Zephyr's __errno.

File: ext/hal/ti/simplelink/source/ti/drivers/net/wifi/porting/user.h [hunk 1]
```diff
@@ -664,7 +666,19 @@ typedef signed int _SlFd_t;
     \warning
 */
 #ifndef SL_INC_INTERNAL_ERRNO
-#define slcb_SetErrno
+/*
+ * Zephyr Port: use Zephyr SDK's errno.h definitions, and supply those missing
+ * to allow the SimpleLink driver.c to compile
+ * Also, supply the external errno setter function.
+ */
+#include <errno.h>
```
Comment (galak) [inline ext/hal/ti/simplelink/source/ti/drivers/net/wifi/porting/user.h]: is there any ability to have this change be in its own file?  Trying to limit changes to imported files if at all possible.

**Review by galak — CHANGES_REQUESTED** [review]
Can you rebase this on latest tree.

File: ext/hal/ti/simplelink/source/ti/drivers/net/wifi/porting/user.h (hunk 1 above)
Comment (GAnthony) [inline ext/hal/ti/simplelink/source/ti/drivers/net/wifi/porting/user.h]: Actually, user.h is one of the files required to change for porting SimpleLink to another OS.  That's why it's (in latest SDK releases) placed in the wifi/**porting/** directory.

The goal is to avoid changing another imported file (driver.c), which needs to include the OS's BSD errno definition when SimpleLink's internal errno mechanism is not used.   Those changes are allowed to be made in the user.h header file (in the wifi/porting/ directory), which gets included in simplelink.h, which driver.c includes.

From driver.c:
#ifdef SL_INC_INTERNAL_ERRNO
#include <ti/drivers/net/wifi/bsd/errno.h>
#endif

So, one alternative would be to modify driver.c directly to include Zephyr's errno.h, but that is not allowed.  I'm not sure there is a different alternative.

I think this change is in the proper file.

Comment (galak) [inline ext/hal/ti/simplelink/source/ti/drivers/net/wifi/porting/user.h]: ok, works for me.

Comment (GAnthony) [inline ext/hal/ti/simplelink/source/ti/drivers/net/wifi/porting/user.h]: However, it would make sense to place all wifi/porting/user.h changes for the port in a separate *commit*, to take it out of the import commit.

Comment (galak) [inline ext/hal/ti/simplelink/source/ti/drivers/net/wifi/porting/user.h]: Yeah, that would be helpful, having that commit message comment about user.h being expected to be modified for OS adaption would be good.

**Review by dbkinder — DISMISSED** [review]
+1 for docs, but there are merge conflicts to address

File: ext/hal/ti/simplelink/source/ti/drivers/net/wifi/porting/user.h (hunk 1 above)
Comment (GAnthony) [inline ext/hal/ti/simplelink/source/ti/drivers/net/wifi/porting/user.h]: OK, rebased and split errno changes into a separate commit.
Thanks @galak for the review.

**Comment by galak** [comment]
Can you rebase, also do we want to update to SDK/HAL version 1_40_01_00?

**Comment by GAnthony** [comment]
Rebased.

I reviewed SDK 1.40.01.00 vs 1.40.00.03 and found no significant functionality differences.

Given the effort to update the SDK file subsets, and the fact this PR is blocking other work, I'd prefer this get merged now, and we update the SDK/HAL versions later if need arises.
